releasenotes/notes/singletons-83782de8bd062cbc.yaml

```diff
@@ -52,6 +52,55 @@ features:
     all the constructor defaults, they will all share a single global
     instance. This results in large reduction in the memory overhead for > 1
     object of these types and significantly faster object construction time.
+  - |
+    Introduced a new class :class:`~.SingletonControlledGate` which is a subclass of
+    :class:`~.ControlledGate` that uses a single instance for all objects of that type.
+    The intent behind this class is to minimize the memory and construction
+    overhead of using multiple gates in a circuit with the tradeoff of having
+    global shared state. For this reason this class is only applicable to
+    gates that do not have any unique and/or mutable state stored in an instance.
+    For example, a :class:`.CXGate` doesn't contain any state and thus can
+    leverage :class:`~.SingletonControlledGate` (and does starting in
+    this release). In contrast, :class:`~.CRXGate` stores an angle parameter as
+    part of its instance data and thus can not use :class:`~.SingletonControlledGate`.
+
+    The other potential issue to be aware of when using
+    :class:`~.SingletonControlledGate` is that the original data model
+    of :class:`~.ControlledGate` supports mutation. Specifically, the
+    :attr:`~.ControlledGate.label`, :attr:`~.ControlledGate.duration`,
+    :attr:`~.ControlledGate.unit`, :attr:`~.ControlledGate.condition`, and
+    :attr:`~.ControlledGate.ctrl_state` attributes are all accessible and mutable in the
+    :class:`~.ControlledGate` and its direct subclasses, but mutation of these attributes
+    on :class:`~.SingletonControlledGate` and its subclasses is disallowed, and will raise
+    and exception. These attributes can be customized but only at creation time (i.e. via
```

Comment:
This isn't quite right anymore since now `SingletonControlledGate` is a direct subclass of `ControlledGate` :3

Reply:
I updated this in: https://github.com/Qiskit/qiskit/pull/10898/commits/3a108ab0b9c769d1899e01b0215740fc46e6aa38 let me know if there is more to update there.